validator/client/src/main/java/tech/pegasys/teku/validator/client/ForkProvider.java

```diff
@@ -43,49 +43,63 @@ public ForkProvider(
     this.asyncRunner = asyncRunner;
     this.validatorApiChannel = validatorApiChannel;
     this.genesisDataProvider = genesisDataProvider;
+    currentFork = new SafeFuture<>();
```

Comment:
nit: move this initialiser to the field declaration.

Reply:
Not sure if you missed this or actually disagree, but typically this would be in the field declaration. :) Main reason is then you can see very clearly that it's not initially null.

Reply:
nope that's me having to manually merge and i missed it, sorry